packages/repl/src/lib/workers/bundler/index.js

```diff
@@ -107,9 +107,63 @@ function has_loopGuardTimeout_feature() {
 	return compare_to_version(3, 14, 0) >= 0;
 }
 
+async function resolve_from_pkg(pkg, subpath) {
+	// match legacy Rollup logic — pkg.svelte takes priority over pkg.exports
+	if (typeof pkg.svelte === 'string' && subpath === '.') {
+		return pkg.svelte;
+	}
+
+	// modern
+	if (pkg.exports) {
+		try {
+			const [resolved] = resolve.exports(pkg, subpath, {
+				browser: true,
+				conditions: ['svelte', 'production']
+			});
+
+			return resolved;
+		} catch {
+			throw `no matched export path was found in "${pkg_name}/package.json"`;
+		}
+	}
+
+	// legacy
+	if (subpath === '.') {
+		const resolved_id = resolve.legacy(pkg, {
+			fields: ['browser', 'module', 'main']
+		});
+
+		if (!resolved_id) {
+			// last ditch — try to match index.js/index.mjs
+			for (const index_file of ['index.mjs', 'index.js']) {
+				try {
+					const indexUrl = new URL(index_file, `${pkg_url_base}/`).href;
+					return await follow_redirects(indexUrl, uid);
+				} catch {
+					// maybe the next option will be successful
+				}
+			}
+
+			throw `could not find entry point in "${pkg_name}/package.json"`;
+		}
+
+		return resolved_id;
+	}
+
+	if (typeof pkg.browser === 'object') {
+		// this will either return `pkg.browser[subpath]` or `subpath`
```

Comment:
@Rich-Harris How did you understand by the [browser option docs](https://github.com/lukeed/resolve.exports#optionsbrowser-1) that if no match on "browser", then `resolve.legacy(pkg, { browser: subpath });` is a relative string path?
I think that in this case it will be `undefined`.

Reply:
You can try it in the Node REPL:

```js
resolve = require('resolve.exports');
resolve.legacy(
  {},
  { browser: './foo.js' }
); // undefined

resolve.legacy(
  { browser: {} },
  { browser: './foo.js' }
); // './foo.js'

resolve.legacy(
  { browser: { './foo.js': './bar.js' } },
  { browser: './foo.js' }
); // './bar.js'
```

Reply:
Make sense, knowing this behavior the docs are now less ambiguous on a second glance.